pkg/controller/atlasdeployment/searchindexes_test.go

```diff
@@ -0,0 +1,662 @@
+//			},
+//		},
+//		{
+//			name: "Shouldn't find indices to delete with equal indices in Atlas and in AKO (exclusive right join)",
+//			args: args{
+//				akoIndices: []*searchindex.SearchIndex{
+//					{SearchIndex: akov2.SearchIndex{Name: "1"}},
+//					{SearchIndex: akov2.SearchIndex{Name: "3"}},
+//				},
+//				atlasIndices: []*searchindex.SearchIndex{
+//					{SearchIndex: akov2.SearchIndex{Name: "1"}},
+//					{SearchIndex: akov2.SearchIndex{Name: "3"}},
+//				},
+//				intersection: ToDelete,
+//			},
+//			want: nil,
+//		},
+//		{
+//			name: "Shouldn't find indices to delete with equal indices in Atlas and in AKO (exclusive right join)",
+//			args: args{
+//				akoIndices: []*searchindex.SearchIndex{
+//					{SearchIndex: akov2.SearchIndex{Name: "1"}},
+//					{SearchIndex: akov2.SearchIndex{Name: "3"}},
+//				},
+//				atlasIndices: []*searchindex.SearchIndex{
+//					{SearchIndex: akov2.SearchIndex{Name: "1"}},
+//					{SearchIndex: akov2.SearchIndex{Name: "3"}},
+//				},
+//				intersection: ToDelete,
+//			},
+//			want: nil,
+//		},
+//		{
+//			name: "Shouldn't find indices to delete with no indices in Atlas (exclusive right join)",
+//			args: args{
+//				akoIndices: []*searchindex.SearchIndex{
+//					{SearchIndex: akov2.SearchIndex{Name: "1"}},
+//					{SearchIndex: akov2.SearchIndex{Name: "3"}},
+//				},
+//				atlasIndices: nil,
+//				intersection: ToDelete,
+//			},
+//			want: nil,
+//		},
+//		{
+//			name: "Shouldn't find indices to delete with no indices in AKO (exclusive right join)",
+//			args: args{
+//				akoIndices:   nil,
+//				atlasIndices: nil,
+//				intersection: ToDelete,
+//			},
+//			want: nil,
+//		},
+//	}
+//	for _, tt := range tests {
+//		t.Run(tt.name, func(t *testing.T) {
+//			assert.Equalf(t, tt.want, findIndexesIntersection(tt.args.akoIndices, tt.args.atlasIndices, tt.args.intersection), "findIndexesIntersection(%v, %v, %v)", tt.args.akoIndices, tt.args.atlasIndices, tt.args.intersection)
+//		})
+//	}
+//}
+
+func Test_getIndexesFromDeploymentStatus(t *testing.T) {
+	tests := []struct {
+		name             string
+		deploymentStatus status.AtlasDeploymentStatus
+		want             map[string]string
+	}{
+		{
+			name: "Should return valid indexes for some valid indexes in the status",
+			deploymentStatus: status.AtlasDeploymentStatus{
+				SearchIndexes: []status.DeploymentSearchIndexStatus{
+					{
+						Name:    "FirstIndex",
+						ID:      "1",
+						Status:  "",
+						Message: "",
+					},
+					{
+						Name:    "SecondIndex",
+						ID:      "2",
+						Status:  "",
+						Message: "",
+					},
+				},
+			},
+			want: map[string]string{
+				"FirstIndex":  "1",
+				"SecondIndex": "2",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, getIndexesFromDeploymentStatus(tt.deploymentStatus), "getIndexesFromDeploymentStatus(%v)", tt.deploymentStatus)
+		})
+	}
+}
+
+//nolint:dupl
```

Comment:
curious: why's that?

Reply:
The linter is a bit dramatic on repeated SearchIndex structures